packages/ra-core/src/controller/list/useList.ts

```diff
@@ -91,12 +91,45 @@ export const useList = <RecordType extends RaRecord = any>(
         total: data ? data.length : undefined,
     }));
 
+    // Store pagination states for each storeKey
+    const storeKeyPaginationRef = useRef<{
+        [key: string]: { page: number; perPage: number };
+    }>({});
+
     // pagination logic
     const { page, setPage, perPage, setPerPage } = usePaginationState({
         page: initialPage,
         perPage: initialPerPage,
     });
 
+    useEffect(() => {
+        if (!resource) return;
+        // Check if storeKey exists in the pagination store
+        const currentPagination = storeKeyPaginationRef.current[resource];
+        if (currentPagination) {
+            // Restore existing pagination state for the storeKey
+            if (
+                page !== currentPagination.page ||
+                perPage !== currentPagination.perPage
+            ) {
+                setPage(currentPagination.page);
+                setPerPage(currentPagination.perPage);
+            }
+        } else {
+            setPage(initialPage);
+            setPerPage(initialPerPage);
+        }
+        storeKeyPaginationRef.current[resource] = { page, perPage };
+    }, [
+        resource,
+        setPage,
+        setPerPage,
+        initialPage,
+        initialPerPage,
+        page,
+        perPage,
+    ]);
+
```

Comment:
I'm afraid you are going in the wrong direction here :grimacing:  . The fix for #10382 suggested by @fzaninotto was to allow to override the `storeKey` used to store the **selection state** (i.e. used in `useRecordSelection`), but not the **pagination** state.

You can have a look at what was done in `useReferenceManyFieldController` for an example:

https://github.com/marmelab/react-admin/blob/7b319c451d448fe9db00b76fa7ddcc10d4d6dedd/packages/ra-core/src/controller/field/useReferenceManyFieldController.ts#L90-L92